server/api/v1/auth/signin/simple.post.ts

```diff
@@ -38,24 +43,55 @@ export default defineEventHandler(async (h3) => {
       statusCode: 401,
       statusMessage: "Invalid username or password.",
     });
-
-  const credentials = authMek.credentials as JsonArray;
-  const hash = credentials.at(1);
-
-  if (!hash || !authMek.user.enabled)
+  else if (!authMek.user.enabled)
     throw createError({
       statusCode: 403,
       statusMessage:
         "Invalid or disabled account. Please contact the server administrator.",
     });
 
-  if (!(await checkHash(password, hash.toString())))
-    throw createError({
-      statusCode: 401,
-      statusMessage: "Invalid username or password.",
-    });
+  // if using old auth schema
+  if (Array.isArray(authMek.credentials)) {
+    const hash = authMek.credentials.at(1);
+
+    if (!hash)
+      throw createError({
+        statusCode: 403,
+        statusMessage:
+          "Invalid password state. Please contact the server administrator.",
+      });
+
+    if (!(await checkHashBcrypt(password, hash.toString())))
+      throw createError({
+        statusCode: 401,
+        statusMessage: "Invalid username or password.",
+      });
+
+    // TODO: send user to forgot password screen or something to force them to change their password to new system
+    await sessionHandler.setUserId(h3, authMek.userId, rememberMe);
+    return { result: true, userId: authMek.userId };
+  } else {
+    // using new (modern) login flow
+
+    const creds = simpleAuth(authMek.credentials);
+    if (creds instanceof type.errors) {
+      // hover out.summary to see validation errors
+      console.error(creds.summary);
+
+      throw createError({
+        statusCode: 400,
+        statusMessage: creds.summary,
```

Comment:
Is this safe to return to the user? This is an internal error because the authMek is invalid.

Reply:
woops, copied this, yea no we definitely shouldn't here